docs/proposals/helm-charts.md

```diff
@@ -0,0 +1,169 @@
+# Helm Chart Registry Adapter Proposal
+
+## Introduction
+
+Add a new registry adapter type, `helm`, that supports Helm chart
+repositories. With this change, an admin, could configure the broker by adding
+a `helm` type registry and have the Helm charts hosted in that Helm chart
+repository converted into objects the broker knows how to manage for use via
+the Service Catalog.
+
+## Problem Description
+
+Currently, our broker only supports docker container image type registries.
+However, with [Helm](https://helm.sh) being the "package manager for Kubernetes",
+there is already a strong collection of applications defined in the form of
+Helm charts. Adding the ability to the broker to consume these Helm charts as
+Service Bundles allows work already done on Helm charts to be made available in
+a cluster.
+
+## Phase 1 - Adding a Basic Helm Chart Registry Adapter
+
+### Creating a Base Image for Executing Helm Charts
+
+Ansible Playbook Bundles (APBs) have already been described as an instance of
+the more generic Service Bundle. Here, we will be creating a new type of
+Service Bundle, Helm Bundles, that respect the Service Bundle contract but use
+Helm as the runtime. We already have a [Helm Bundle
+Base](https://github.com/ansibleplaybookbundle/helm-bundle-base) and the
+changes shown in [Helm Bundle Base PR 2](https://github.com/ansibleplaybookbundle/helm-bundle-base/pull/2)
+show the work required to implement the Service Bundle contract using Helm.
+
+### Broker Config Options
+
+It is expected that a broker admin who wants to expose Helm charts would update
+the broker's config to look something like:
+
+```
+registry:
+  - type: helm
+    name: stable
+    url: "https://kubernetes-charts.storage.googleapis.com"
+    base_image: "docker.io/djzager/helm-bundle-base:latest"
+    white_list:
+      - ".*"
+```
+
+1. Type: refers to the registry adapter to handle this registry
+1. Name: gives a name to this registry item
+1. URL: the URL for the Helm chart repository
+1. Base Image: the container image to use when provisioning/deprovisioning a
+   Helm chart
+1. White List/Black List: allows the Admin to filter out Helm charts
+
+#### Changes to the Registry Package
+
+- Add the ability to read the `base_image` from the broker config.
+- Pass the URL and BaseImage to the registry adapter.
+- Instantiate a Helm registry adapter to handle Helm Chart registries in the
+  config.
+
+#### Creating a Helm Adapter
+
+The `HelmAdapter` struct will look something like, the addition of `Charts`
+makes it possible to save our work and only ready the registries index file
+once:
+
+```
+// import "k8s.io/helm/pkg/repo"
+// HelmAdapter - Helm Registry Adapter
+type HelmAdapter struct {
+    Config Configuration
+    Log    *logging.Logger
+    Charts map[string]*repo.ChartVersion
+}
+```
```

Comment:
Let's reword this section. I had to read it twice to understand what was going on. 

The `HelmAdapter` struct will look something like the example below:

```golang
// import "k8s.io/helm/pkg/repo"
// HelmAdapter - Helm Registry Adapter
type HelmAdapter struct {
    Config Configuration
    Log    *logging.Logger
    Charts map[string]*repo.ChartVersion
}
```
The addition of the `Charts` field make it possible to save our work and only read the registries' index file once.